ci/scripts/setup-vault.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+echo "Setting up HashiCorp Vault test data..."
+
```

Comment:
Are there some docs from hashicorp for the setup steps? We can link them.

Reply:
updated in code comment.